src/genconstants.c

```diff
@@ -0,0 +1,296 @@
+	PCONST(i32, KEY_SPREVIOUS);
+	PCONST(i32, KEY_SPRINT);
+	PCONST(i32, KEY_SREDO);
+	PCONST(i32, KEY_SREPLACE);
+	PCONST(i32, KEY_SRIGHT);
+	PCONST(i32, KEY_SRSUME);
+	PCONST(i32, KEY_SSAVE);
+	PCONST(i32, KEY_SSUSPEND);
+	PCONST(i32, KEY_SUNDO);
+	PCONST(i32, KEY_SUSPEND);
+	PCONST(i32, KEY_UNDO);
+	PCONST(i32, KEY_MOUSE);
+	PCONST(i32, KEY_RESIZE);
+#ifdef KEY_EVENT
+	PCONST(i32, KEY_EVENT);
+#endif
+	PCONST(i32, KEY_MAX);
+
+#ifdef NCURSES_MOUSE_VERSION
+	PCONST(i32, NCURSES_MOUSE_VERSION);
+#endif
+
+#ifdef MASK_SHIFT
+	PCONST(i32, MASK_SHIFT);
+#endif
+
+#ifdef MODIFIER_SHIFT
+	PCONST(i32, MODIFIER_SHIFT);
+#endif
+
+	/* Mouse Support */
+#ifdef NCURSES_BUTTON_RELEASED
+	PCONST(i32, NCURSES_BUTTON_RELEASED);
+#endif
+
+#ifdef NCURSES_BUTTON_PRESSED
+	PCONST(i32, NCURSES_BUTTON_PRESSED);
+#endif
+
+#ifdef NCURSES_BUTTON_CLICKED
+	PCONST(i32, NCURSES_BUTTON_CLICKED);
+#endif
+
+#ifdef NCURSES_DOUBLE_CLICKED
+	PCONST(i32, NCURSES_DOUBLE_CLICKED);
+#endif
+
+#ifdef NCURSES_TRIPLE_CLICKED
+	PCONST(i32, NCURSES_TRIPLE_CLICKED);
+#endif
+
+#ifdef NCURSES_RESERVED_EVENT
+	PCONST(i32, NCURSES_RESERVED_EVENT);
+#endif
+
+	/* event masks */
+	PCONST(i32, BUTTON1_RELEASED);
+	PCONST(i32, BUTTON1_PRESSED);
+	PCONST(i32, BUTTON1_CLICKED);
+	PCONST(i32, BUTTON1_DOUBLE_CLICKED);
+	PCONST(i32, BUTTON1_TRIPLE_CLICKED);
+
+	PCONST(i32, BUTTON2_RELEASED);
+	PCONST(i32, BUTTON2_PRESSED);
+	PCONST(i32, BUTTON2_CLICKED);
+	PCONST(i32, BUTTON2_DOUBLE_CLICKED);
+	PCONST(i32, BUTTON2_TRIPLE_CLICKED);
+
+	PCONST(i32, BUTTON3_RELEASED);
+	PCONST(i32, BUTTON3_PRESSED);
+	PCONST(i32, BUTTON3_CLICKED);
+	PCONST(i32, BUTTON3_DOUBLE_CLICKED);
+	PCONST(i32, BUTTON3_TRIPLE_CLICKED);
+
+	PCONST(i32, BUTTON4_RELEASED);
+	PCONST(i32, BUTTON4_PRESSED);
+	PCONST(i32, BUTTON4_CLICKED);
+	PCONST(i32, BUTTON4_DOUBLE_CLICKED);
+	PCONST(i32, BUTTON4_TRIPLE_CLICKED);
+
+#ifdef BUTTON5_RELEASED
+	PCONST(i32, BUTTON5_RELEASED);
+#endif
+
+#ifdef BUTTON5_PRESSED
+	PCONST(i32, BUTTON5_PRESSED);
+#endif
+
+#ifdef BUTTON5_CLICKED
+	PCONST(i32, BUTTON5_CLICKED);
+#endif
+
+#ifdef BUTTON5_DOUBLE_CLICKED
+	PCONST(i32, BUTTON5_DOUBLE_CLICKED);
+#endif
+
+#ifdef BUTTON5_TRIPLE_CLICKED
+	PCONST(i32, BUTTON5_TRIPLE_CLICKED);
+#endif
+
+	PCONST(i32, BUTTON_CTRL);
+	PCONST(i32, BUTTON_SHIFT);
+	PCONST(i32, BUTTON_ALT);
+	PCONST(i32, REPORT_MOUSE_POSITION);
+
+	PCONST(i32, ALL_MOUSE_EVENTS);
+
+	/* Attributes */
+	PCONSTU(crate::ll::chtype, A_NORMAL);
+	PCONSTU(crate::ll::chtype, A_STANDOUT);
+	PCONSTU(crate::ll::chtype, A_UNDERLINE);
+	PCONSTU(crate::ll::chtype, A_REVERSE);
+	PCONSTU(crate::ll::chtype, A_BLINK);
+	PCONSTU(crate::ll::chtype, A_DIM);
+	PCONSTU(crate::ll::chtype, A_BOLD);
+
+#ifdef A_BLANK
+	PCONSTU(crate::ll::chtype, A_BLANK);
+#endif
+
+	PCONSTU(crate::ll::chtype, A_INVIS);
+	PCONSTU(crate::ll::chtype, A_PROTECT);
+	PCONSTU(crate::ll::chtype, A_ALTCHARSET);
+	PCONSTU(crate::ll::chtype, A_ATTRIBUTES);
+	PCONSTU(crate::ll::chtype, A_CHARTEXT);
+	PCONSTU(crate::ll::chtype, A_COLOR);
+}
```

Comment:
Missing `A_ITALIC` ? I can't tell if it was on purpose...

`A_ITALIC` was here before:
https://github.com/jeaye/ncurses-rs/pull/201/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L1755-L1756